src/gasPriceOracle/oracle.ts

```diff
@@ -0,0 +1,92 @@
+import { BigNumber, providers } from "ethers";
+import { toBN } from "../utils";
+
+export type GasPriceEstimate = {
+  maxFeePerGas: BigNumber;
+  maxPriorityFeePerGas?: BigNumber;
+};
+
+type ValidatedFeeData = {
+  gasPrice: BigNumber;
+  maxFeePerGas: BigNumber;
+  maxPriorityFeePerGas: BigNumber;
+};
+
+interface GasPriceFeed {
+  (provider: providers.Provider): Promise<GasPriceEstimate>;
+}
+
+async function getProviderFeeData(provider: providers.Provider): Promise<ValidatedFeeData> {
+  const validateFeeData = (val: BigNumber | null): BigNumber  => {
+    return BigNumber.isBigNumber(val) && val.gt(0) ? val : toBN(Number.MAX_SAFE_INTEGER);
+  };
+
+  // Not much to do on a failed retrieval, so just suppress it.
+  const response: providers.FeeData = await provider.getFeeData().catch(() => {
+    return {} as providers.FeeData
+  });
+
+  const feeData: ValidatedFeeData = {
+    "gasPrice": validateFeeData(response["gasPrice"]),
+    "maxFeePerGas": validateFeeData(response["maxFeePerGas"]),
+    "maxPriorityFeePerGas": validateFeeData(response["maxPriorityFeePerGas"]),
```

Comment:
Does this mean that if tat RPC call fails, we default to an extremely high gas price? What are the implications of that? For the FE, would we want to show a crazy high fee or would we prefer to have the call fail and just fail to compute? What about other callers? I feel like throwing (unless we have a retry or fallback) would give the caller the most flexibility, no?

Reply:
> Does this mean that if tat RPC call fails, we default to an extremely high gas price? What are the implications of that? For the FE, would we want to show a crazy high fee or would we prefer to have the call fail and just fail to compute? What about other callers? I feel like throwing (unless we have a retry or fallback) would give the caller the most flexibility, no?

All good questions :) Some of them were raised (without conclusion) here: https://github.com/across-protocol/sdk-v2/pull/102#discussion_r982098175

My current thinking is to live with the status quo (i.e. don't throw on price lookup failure; default instead to a price of 0).

Implication:
 - Callers need to specifically handle that a price of 0 might be returned. Existing Across "users" of this code do handle this correctly.

Throwing an exception here would probably be the _most_ correct thing to do, but require that callers handle that, and none currently do. Are we interested in upgrading them so that they can tolerate an exception?

Reply:
I think we should throw if the RPC call fails (after retry) rather than catching and converting to some fallback value.

In terms of the fee data _shape_, that _could_ differ by chain. Are we covering all the chains we support? If so, shouldn't we just throw an error if something doesn't match our expectations, since that would imply there is a bug in our code, no?